src/core/Group.cpp

```diff
@@ -552,59 +548,47 @@ QList<Entry*> Group::entriesRecursive(bool includeHistoryItems) const
     return entryList;
 }
 
-Entry* Group::findEntry(QString entryId)
+Entry* Group::findEntryByUuid(const QUuid& uuid) const
 {
-    Q_ASSERT(!entryId.isNull());
-
-    Entry* entry;
-    QUuid entryUuid = QUuid::fromRfc4122(QByteArray::fromHex(entryId.toLatin1()));
-    if (!entryUuid.isNull()) {
-        entry = findEntryByUuid(entryUuid);
-        if (entry) {
-            return entry;
-        }
-    }
-
-    entry = findEntryByPath(entryId);
-    if (entry) {
-        return entry;
-    }
-
-    for (Entry* entry : entriesRecursive(false)) {
-        if (entry->title() == entryId) {
-            return entry;
+    if (!uuid.isNull()) {
```

Comment:
```
if (uuid.isNull()) {
    return nullptr;
}
```

Reply:
Why? then this function would have three return statements (short circuit, in-loop, and fall-through). I don't agree with that approach.